assets/js/osmenu.js

```diff
@@ -1,30 +1,31 @@
 $(document).ready(function(){
-  $('#operatingSystems').change(function() {
-    switch($(this).val()) {
-      case "rhel6":
-        $('#rhel6').show();
-        $('#rhel7').hide();
-        $('#el6').show();
-        $('#el7').hide();
-      break;
-      case "rhel7":
-        $('#rhel6').hide();
-        $('#rhel7').show();
-        $('#el6').hide();
-        $('#el7').show();
-      break;
-      case "el6":
-        $('#rhel6').hide();
-        $('#rhel7').hide();
-        $('#el6').show();
-        $('#el7').hide();
-        break;
-      case "el7":
-        $('#rhel6').hide();
-        $('#rhel7').hide();
-        $('#el6').hide();
-        $('#el7').show();
-        break;
-    }
-  });
+    $('#operatingSystems').change(function() {
+        var operatingSystems = [
+            'rhel5',
+            'rhel6',
+            'rhel7',
+            'el5',
+            'el6',
+            'el7',
+            'fc20',
+            'fc21'
+        ];
+
+        function showOS(operatingSystem) {
+            var i = 0,
+                length = operatingSystems.length;
+
+            for(i; i < length; i += 1) {
+                if (operatingSystems[i] === operatingSystem) {
+                    $('#' + operatingSystem).show();
+                } else {
+                    $('#' + operatingSystems[i]).hide();
+                }
+            }
+
+            return true;
+        }
+
+        showOS($(this).val());
+    });
 });
```

Comment:
Fancier Javascript :+1: 

But the logic is slightly different, if you'll notice rhel6 enables both rhel6 and el6 on the install page, the new javascript doesn't do that.